src/main/java/it/eng/idsa/dataapp/handler/ArtifactMessageHandler.java

```diff
@@ -30,15 +34,23 @@
 @Component
 public class ArtifactMessageHandler extends DataAppMessageHandler {
 
-	@Value("#{new Boolean('${application.encodePayload:false}')}")
 	private Boolean encodePayload;
-
 	private SelfDescriptionService selfDescriptionService;
+	private ThreadService threadService;
```

Comment:
Overall comment: can you put methods in the way how they are called?
Maybe mine OCD is taking over :D